src/imperative/batching.cc

```diff
@@ -132,14 +132,16 @@ nnvm::Graph DBatchEngine::BatchGraphs(const std::vector<nnvm::Graph>& graphs) {
   // collect depth
   int max_depth = 0;
   std::unordered_map<nnvm::Node*, int> depth_map;
-  std::vector<nnvm::NodeEntry> prev_outputs;
+  std::vector<nnvm::NodeEntry> prev_outputs, all_old_entries;
   for (const nnvm::Graph& g : graphs) {
     const std::vector<nnvm::NodeEntry>& g_outputs = g.outputs;
     LOG(INFO) << "graph outputs " << g_outputs.size();
     std::copy(g_outputs.begin(), g_outputs.end(), std::inserter(prev_outputs, prev_outputs.end()));
+    std::copy(g_outputs.begin(), g_outputs.end(), std::inserter(all_old_entries, all_old_entries.end()));
     nnvm::DFSVisit(g_outputs, [&](const nnvm::NodePtr& n){
       int depth = 0;
       for (auto e : n->inputs) {
+        all_old_entries.emplace_back(e);
         int idepth = depth_map.at(e.node.get());
         depth = std::max(depth, idepth+1);
       }
```

Comment:
line 174 should be a reference instead of copy

Reply:
line 171: we should consider making an alias for this type `std::unordered_map<std::vector<nnvm::NodeEntry>, nnvm::NodeEntry, NodeEntryVectorHash, NodeEntryVectorEqual> ` just like NodeEntryMap.. It's too long